### ISSUE TITLE
[savepanel] doesn't remember last file location

### ISSUE THREAD

**Issue by Spacechild1** [pure-data/pure-data#1018]
unlike `[openpanel]`, `[savepanel]` doesn't remember the last file location, i. e. it will always start in the home directory resp. current working directory which can be quite annoying.

tested on Windows 7 with Pd 0.50-2 and latest master (285c53ff57621801e794d1e0c13425193bde3d61)

**Comment by umlaeute** [comment]
these days you can build this with:

```
[bang(
|        +----------------+
[symbol .]                |
|                         |
[savepanel]               |
|                         |
[t a a]                   |
|     [file splitname]    |
|     [unpack s]          |
|     |                   |
|     +-------------------+
|
```

is this good enough?

**Comment by Spacechild1** [comment]
> is this good enough?

IMO, no. Remembering the last save location is a standard feature on every desktop app. Users shouldn't need to build this themselves.

**Comment by danomatika** [comment]
It should remember. Check the main proc in wheredoesthisgo.tcl and it should set it there when the panel is made.enohp ym morf tnes-----------Dan Wilcoxdanomatika.comrobotcowboy.comOn Dec 14, 2023, at 10:28 AM, Christof Ressi ***@***.***> wrote:﻿

is this good enough?

IMO, no. Remembering the last save location is a standard feature on every desktop app. Users shouldn't need to build this themselves.

**Comment by Spacechild1** [comment]
> It should remember. Check the main proc in wheredoesthisgo.tcl and it should set it there when the panel is made.

With "should" do mean "it already does it" or "this needs to be done"?

At least with Pd 0.54.1, `[savepanel]` does indeed start in the folder of the most recently saved *patch*, but it does not update the path itself (which it really should).

**Comment by danomatika** [comment]
I don't remember exactly. I updated the default path handling to use the Pd do edit when the plugin is enabled. At the time I think I wanted to double check that it worked "as expected" like  stove dialogs but I think I ran out of time.My suggestion is to double check, really. Check what the proc does and where it is called in the tcl. That would be helpful.enohp ym morf tnes-----------Dan Wilcoxdanomatika.comrobotcowboy.comOn Dec 14, 2023, at 10:44 AM, Christof Ressi ***@***.***> wrote:﻿

It should remember. Check the main proc in wheredoesthisgo.tcl and it should set it there when the panel is made.

With "should" do mean "it already does it" or "this needs to be done"?
At least with Pd 0.54.1, [savepanel] does indeed start in the folder of the most recently saved patch, but it does not update the path itself (which it really should).
